protocol/x/rewards/module_test.go

```diff
@@ -0,0 +1,145 @@
+	interfaceRegistry := types.NewInterfaceRegistry()
+	appCodec := codec.NewProtoCodec(interfaceRegistry)
+
+	appModule := rewards.NewAppModuleBasic(appCodec)
+	require.NotNil(t, appModule)
+
+	return appModule
+}
+
+func TestAppModuleBasic_RegisterCodecLegacyAmino(t *testing.T) {
+	am := createAppModuleBasic(t)
+
+	cdc := codec.NewLegacyAmino()
+	am.RegisterLegacyAminoCodec(cdc)
+
+	var buf bytes.Buffer
+	err := cdc.Amino.PrintTypes(&buf)
+	require.NoError(t, err)
+}
+
+func TestAppModuleBasic_ValidateGenesisErr(t *testing.T) {
+	tests := map[string]struct {
+		genesisJson string
+		expectedErr string
+	}{
+		"Invalid json": {
+			genesisJson: `{"missingClosingQuote: true}`,
+			expectedErr: "failed to unmarshal rewards genesis state: unexpected EOF",
+		},
+		"Invalid state": {
+			genesisJson: `{"params":{}}`,
+			expectedErr: "treasury account cannot have empty name: invalid treasury account",
+		},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			am := createAppModuleBasic(t)
+
+			interfaceRegistry := types.NewInterfaceRegistry()
+			cdc := codec.NewProtoCodec(interfaceRegistry)
+
+			err := am.ValidateGenesis(cdc, nil, json.RawMessage(tc.genesisJson))
+			require.EqualError(t, err, tc.expectedErr)
+		})
+	}
+}
+
+func TestAppModuleBasic_RegisterGRPCGatewayRoutes(t *testing.T) {
+	am := createAppModuleBasic(t)
+
+	router := runtime.NewServeMux()
+
+	am.RegisterGRPCGatewayRoutes(client.Context{}, router)
+
+	// Expect NumMessages route registered
+	recorder := httptest.NewRecorder()
+	req, err := http.NewRequest("GET", "/dydxprotocol/v4/rewards/params", nil)
+	require.NoError(t, err)
+	router.ServeHTTP(recorder, req)
+	require.Contains(t, recorder.Body.String(), "no RPC client is defined in offline mode")
+
+	// Expect unexpected route not registered
+	recorder = httptest.NewRecorder()
+	req, err = http.NewRequest("GET", "/dydxprotocol/v4/rewards/foo/bar/baz", nil)
+	require.NoError(t, err)
+	router.ServeHTTP(recorder, req)
+	require.Equal(t, 404, recorder.Code)
+}
+
+func TestAppModuleBasic_GetTxCmd(t *testing.T) {
+	am := createAppModuleBasic(t)
+
+	cmd := am.GetTxCmd()
+	require.Equal(t, "rewards", cmd.Use)
+	require.Equal(t, 0, len(cmd.Commands()))
+}
+
+func TestAppModuleBasic_GetQueryCmd(t *testing.T) {
+	am := createAppModuleBasic(t)
+
+	cmd := am.GetQueryCmd()
+	require.Equal(t, "rewards", cmd.Use)
+	require.Equal(t, 1, len(cmd.Commands()))
+	require.Equal(t, "params", cmd.Commands()[0].Name())
+}
+
+func TestAppModule_InitExportGenesis(t *testing.T) {
+	am, keeper, ctx := createAppModuleWithKeeper(t)
+	interfaceRegistry := types.NewInterfaceRegistry()
+	cdc := codec.NewProtoCodec(interfaceRegistry)
+
+	validGenesisState := pricefeed.ReadJsonTestFile(t, "expected_default_genesis.json")
+
+	gs := json.RawMessage(validGenesisState)
+
+	result := am.InitGenesis(ctx, cdc, gs)
+	require.Equal(t, 0, len(result))
+
+	params := keeper.GetParams(ctx)
+
+	require.Equal(t, "rewards_treasury", params.TreasuryAccount)
```

Comment:
These values need to be updated per recent PRs that changes the default value - see comment below!